### PR TITLE
[Core] Remove OSUtilities

### PR DESCRIPTION
**Pull request by loumalouomega**
Removing from the last places used (@philbucher I was the only one using it, I think it is safe to remove)

### CODE REVIEW

File: applications/CSharpWrapperApplication/tests/cpp_tests/test_calculate.cpp [hunk 1]
```diff
@@ -22,13 +22,13 @@
...
 #include "custom_includes/kratos_wrapper.h"
-#include "utilities/os_utilities.h"
+#include "includes/kratos_filesystem.h"
 
 namespace Kratos {
     namespace Testing {
         void CreateMDPAFile() {
             std::filebuf buffer;
-            buffer.open(OSUtilities::GetCurrentWorkingDir() + "/file.mdpa", std::ios::out);
+            buffer.open(FilesystemExtensions::JoinPaths({FilesystemExtensions::CurrentWorkingDirectory(), "/file.mdpa"}), std::ios::out);
```
Comment (philbucher) [inline applications/CSharpWrapperApplication/tests/cpp_tests/test_calculate.cpp]:
```suggestion
            buffer.open(FilesystemExtensions::JoinPaths({FilesystemExtensions::CurrentWorkingDirectory(), "file.mdpa"}), std::ios::out);
```

`FilesystemExtensions::JoinPaths` adds the separator

**Review by philbucher — COMMENTED** [review]
please remove the separators otherwise 👍

File: applications/CSharpWrapperApplication/tests/cpp_tests/test_calculate.cpp (hunk 1 above)
Comment (loumalouomega) [reply applications/CSharpWrapperApplication/tests/cpp_tests/test_calculate.cpp]: Ups

**Review by loumalouomega — COMMENTED** [review]
Ups

**Review by philbucher — APPROVED** [review]
👍 
Thx

**Pull request merged**